### ISSUE TITLE
Inappropriate kernel memory exposure

### ISSUE THREAD

**Issue by jserv** [jserv/simplefs#10]
Use `make check` to reproduce.

Reported by `dmesg`:
```
Bad or missing usercopy whitelist? Kernel memory exposure attempt detected from SLUB object 'simplefs_cache' (offset 4, size 4)!
Call Trace:
 __check_heap_object+0xb7/0x120
 __check_object_size+0xdb/0x1b0
 readlink_copy+0x4c/0x80
 vfs_readlink+0x7e/0x110
 do_readlinkat+0xf8/0x120
 __x64_sys_readlink+0x1f/0x30
 do_syscall_64+0x5a/0x130
 entry_SYSCALL_64_after_hwframe+0x44/0xa9
```

**Comment by RoyWFHuang** [comment]
Should this issue  be closed? this issue seems the same with #30

**Comment by jserv** [comment]
Confirmed.
